packages/material-ui/src/Typography/Typography.js

```diff
@@ -112,6 +113,34 @@ const defaultVariantMapping = {
   body2: 'p',
 };
 
+const isSupportedColor = (color) => {
+  const supportedColors = [
+    'primary',
+    'secondary',
+    'initial',
+    'inherit',
+    'textPrimary',
+    'textSecondary',
+    'error',
+  ];
+
+  return supportedColors.indexOf(color) >= 0;
+}
+
+const useColorStyles = makeStyles((theme) => ({
+  color: {
+    color: (props) => (theme.palette[props.color] ? theme.palette[props.color].main : 'inherit'),
+  },
+}));
+
+const useDynamicTypographyColor = ({ color }) => {
+  if (!isSupportedColor(color)) {
```

Comment:
I don't think that it can work, it would break the rules of hooks

Reply:
Yep, I forgot that the `makeStyles` returns a hook 👍